### PR TITLE
Inheritance Bequest: Implement internal transaction annotations over the internal provider

### PR DESCRIPTION
**Pull request by Shadowfiend**
This PR does a few things:
- Guards the provider bridge against dApps that try to impersonate the internal provider origin.
- Updates the internal provider service to allow transaction requests from the internal provider origin to pre-populate transaction annotations.
- Updates the internal Ethers provider to allow annotations to be submitted to the internal provider origin.
- Updates the swap async thunk to provide a swap annotation when submitting a swap transaction.

Notably, *only the new signing flow uses this annotation*. The support in the old signing flow for swap annotations is explicitly disabled, as there hasn't been design alignment around the swap page, and the existing implementation is entirely unfinished. Rather than put it behind an independent feature flag, for now it's just behind the new signing flow flag. We may decide to introduce an independent flag if the new signing flow is ready to go but the swap summary is not.

Screen Shot 2022-09-29 at 12 21 41

## Testing

- [ ] With the `USE_UPDATED_SIGNING_UI` flag off, ensure swaps continue to work exactly as-is. In particular, the signature confirmation screen should look like a standard contract interaction.
- [ ] Ensure dApps can continue to submit transactions for signing.
- [ ] With the `USE_UPDATED_SIGNING_UI` flag on, ensure a signature confirmation screen more like the screenshot above appears instead.

See #2210.

### CODE REVIEW

**Comment by mhluongo** [comment]
Haven't reviewed yet, but it'd be awful nice to see that `0xdef1...` properly replaced as "0x Router" :eyes:

**Review by 0xDaedalus — APPROVED** [review]
素晴らしい

**Comment by Shadowfiend** [comment]
Yeah this PR won't resolve that. We either need to do a bigger annotation pass in the swap slice (not my preference) or find a good way to not skip the enrichment service when there is an existing annotation from the provider (my preference, but haven't dug in yet).

**Comment by Shadowfiend** [comment]
I guess also to be clear: I think resolving that should block the UI side of this being merged it. What that means if the new sign transaction flow is deemed ready to go before this part is handled, we would put this bit behind its own feature flag.

File: background/services/internal-ethereum-provider/index.ts [hunk 1]
```diff
@@ -311,6 +319,14 @@ export default class InternalEthereumProviderService extends BaseService<Events>
     transactionRequest: JsonRpcTransactionRequest,
     origin: string
   ): Promise<SignedTransaction> {
+    const annotation =
+      origin === TALLY_INTERNAL_ORIGIN &&
+      "annotation" in transactionRequest &&
+      transactionRequest.annotation !== undefined
+        ? // We use  `as` here as we know it's from a trusted source.
+          (decodeJSON(transactionRequest.annotation) as TransactionAnnotation)
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply background/services/internal-ethereum-provider/index.ts:327]: I did consider that, but with transaction annotations having so many forms, and likely to add more, it seemed like we might sprout a full suite of single-use type guards to check something that is serialized in a call that already has a type check on it. In practice this is likely to be used exclusively through the augmented and type-checked internal provider. 

50-50 on it. The safety monk in me wants to do it, the pragmatist in me says what we have here is sufficient.

**Pull request merged**